utils/contextstore/contextstore.go

```diff
@@ -69,11 +71,20 @@ func (s *store) Retrieve(contextID string, context interface{}) error {
 	}
 
 	data, err := ioutil.ReadFile(filepath.Join(folder, itemFile))
+
 	if err != nil {
 		return fmt.Errorf("unable to retrieve context from store: %s", err)
 	}
 
 	if err = json.Unmarshal(data, context); err != nil {
+		if s.dataErrorHandler != nil {
+			if err = s.dataErrorHandler(string(data), context); err == nil {
+				if err = s.Store(contextID, context); err != nil {
+					return fmt.Errorf("Remap attempt failed with error %s", err)
```

Comment:
In case of error, should we remove or not. Today we remove because now we can keep running this but we will never recover from this error no matter how many restarts we do.